cdap-explore/src/main/java/co/cask/cdap/explore/service/ExploreTableManager.java

```diff
@@ -524,6 +530,19 @@ private String generateFileSetCreateStatement(DatasetId datasetId, Dataset datas
     String databaseName = ExploreProperties.getExploreDatabaseName(properties);
     Map<String, String> tableProperties = FileSetProperties.getTableProperties(properties);
 
+    // if this dataset reuses an existing table, do not attempt to create it
+    if (FileSetProperties.isUseExisting(tableProperties)
+      || FileSetProperties.isPossessExisting(tableProperties)) {
+      try {
+        exploreService.getTableInfo(databaseName, tableName);
+        // table exists: do not attempt to create
+        return null;
+      } catch (Exception e) {
+        LOG.warn("Dataset '{}' is configured to use an existing explore table, but table '{}' does not " +
```

Comment:
an exception doesn't always mean the table doesn't exist does it? Guessing Hive doesn't throw a more specific exception in this scenario?

In any case, is the right behavior really to create it if it doesn't exist? Seems different than the behavior for the directory on hdfs.